magento/pwa-studio
# AC-2480::Flyout panels do not trap focus while open

## Description

Flyout panels do not trap focus while open

Environment
Adobe Magento - Venia

Context
Windows 10; Chrome 88

Reproduction Steps
Locations (representative sample):

Global Header
Search Results
1. Using keyboard only, from a search results page activate the "Filter" button.
2. Press the Tab key to navigate to the Filters panel. Note that focus lands on elements obscured by the modal overlay.

Actual Behavior
The mobile site menu (globally) and Filters panel (on Search/list pages) add an overlay to the rest of the page content when they are expanded, but keyboard focus is not trapped within them while they are open. As a result, keyboard focus lands on elements that are obscured beneath the overlay.

Expected Behavior
When a content panel such as the mobile navigation is open and an overlay obscures other content on the screen, the dialog should be marked up as and behave as a modal dialog.

Ensure that when these panels are open and an overlay obscures the rest of the page content, keyboard focus is trapped within the panel.

Please refer to WAI-ARIA Authoring Practices Dialog (Modal) design pattern example:

https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal



## Related Issue







Closes https://jira.corp.magento.com/browse/AC-2480

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps

✔️ QA Passed
Pre-Conditions:

1. Have Magento instance with sample data installed

2. Make sure to have pwa studio installed

3. Make sure to have a customer login for front end login

 

Manual Steps executed:

Login to venia > Navigate to global header accessories or other
Navigate to main menu and check focus navigating to the options given 
Navigate to  filter button and check focus navigating to the options given 
 

✖️ Behaviour Before The Fix : The keyboard focus is not trapped within the main menu while they are open.

✔️Behaviour After The Fix: The keyboard focus is getting trapped within the main menu while they are open.


## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.


### Resolved issues:
1. [x] resolves magento/pwa-studio#3823: AC-2480::Flyout panels do not trap focus while open
